### PR TITLE
Adding panic-handler and allocator features to `icu_capi`

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]
r=me but I'd also wait for @sffc's approval

**Review by Manishearth — DISMISSED** [review]

File: docs/tutorials/cpp.md
Comment (sffc) [inline docs/tutorials/cpp.md:112]: Question: how does panic-abort and panic-immediate-abort work with this new built-in panic handler?

File: ffi/capi/Cargo.toml
Comment (sffc) [inline ffi/capi/Cargo.toml:111]: Nits on feature names:

1. It seems we're using `_` for feature names so we should be consistent
2. `panic-handler` should be more specific, like `looping_panic_handler`

**Review by sffc — COMMENTED** [review]
Seems good

**Review by sffc — APPROVED** [review]

**Pull request merged**
